File: #0997.find-the-town-judge.js
/**
 * @param {number} n
 * @param {number[][]} trust
 * @return {number}
 */
var findJudge = function(n, trust) {
    let fill = Array(n + 1).fill(0);
    for (let [left, right] of trust) {
        fill[left] = fill[left] - 1;
        fill[right] = fill[right] + 1;
    }

    for (let i = 1; i < fill.length; i++) {
        if (fill[i] == n - 1) {
            return i;
        }
    }
    return -1;
};

File: #0997.find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {

        vector<int> indegrees(n + 1);
        vector<int> outdegrees(n + 1);
        for (auto a : trust) {
            ++outdegrees[a[0]];
            ++indegrees[a[1]];
        }
        for (int i = 1; i <= n; ++i) {
            if (indegrees[i] == n - 1 && outdegrees[i] == 0) {
                return i;
            }
        }
        return -1;
    }
};
